Content.Server/StationEvents/Events/Theta/ShipEvent-FlagCapture.cs

```diff
@@ -0,0 +1,122 @@
+using System.Threading;
+using Content.Server.GameTicking;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Theta.DebrisGeneration;
+using Content.Server.Theta.ShipEvent.Components;
+using Content.Server.Theta.ShipEvent.Systems;
+using Content.Shared.Roles.Theta;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Timer = Robust.Shared.Timing.Timer;
+
+namespace Content.Server.StationEvents.Events.Theta;
+
+
+[RegisterComponent, Access(typeof(SEFCRule))]
+public sealed partial class SEFCRuleComponent : Component { } //yes partial is needed here
+
+[RegisterComponent]
+public sealed partial class SEFCFlagComponent : Component
+{
+    /// <summary>
+    /// Last team which has captured this flag
+    /// </summary>
+    public ShipEventFaction LastTeam = default!;
```

Comment:
```suggestion
    public ShipEventFaction? LastTeam;
```
Почему это нон нуллейбл штука, если по дефолту у флага нулл?

Reply:
ну типо не слишком много смысла от того что она нуллейбл.